features/src/main/scala/com/salesforce/op/stages/base/sequence/BinarySequenceTransformer.scala

```diff
@@ -0,0 +1,140 @@
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.stages.base.sequence
+
+import com.salesforce.op.UID
+import com.salesforce.op.features.FeatureSparkTypes
+import com.salesforce.op.features.types.FeatureType
+import com.salesforce.op.stages.{OpPipelineStage2N, OpTransformer}
+import org.apache.spark.ml.Transformer
+import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.util.ClosureUtils
+import org.apache.spark.sql.functions._
+
+import scala.reflect.runtime.universe.TypeTag
+import scala.util.Try
+
+trait OpTransformer2N[I1 <: FeatureType, I2 <: FeatureType, O <: FeatureType]
+  extends Transformer with OpPipelineStage2N[I1, I2, O] with OpTransformer {
+
+  implicit val tti1: TypeTag[I1]
+  implicit val tti2: TypeTag[I2]
+
+  /**
+   * Function used to convert input to output
+   */
+  def transformFn: (I1, Seq[I2]) => O
+
+  /**
+   * Check if the stage is serializable
+   *
+   * @return Failure if not serializable
+   */
+  final override def checkSerializable: Try[Unit] = ClosureUtils.checkSerializable(transformFn)
+
+  /**
+   * Spark operation on dataset to produce new output feature column using defined function
+   *
+   * @param dataset input data for this stage
+   * @return a new dataset containing a column for the transformed feature
+   */
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    assert(inN.nonEmpty, "Inputs cannot be empty")
+    val newSchema = setInputSchema(dataset.schema).transformSchema(dataset.schema)
+    val functionUDF = FeatureSparkTypes.udf2N[I1, I2, O](transformFn)
+    val meta = newSchema(getOutputFeatureName).metadata
+    val columns = getTransientFeatures().map(in => dataset.col(in.name))
+    dataset.select(col("*"), functionUDF(struct(columns: _*)).as(getOutputFeatureName, meta))
+  }
+
+  private val transformNFn = FeatureSparkTypes.transform2N[I1, I2, O](transformFn)
+  override def transformKeyValue: KeyValue => Any = {
```

Comment:
lazy val

Reply:
I am not sure what should be lazy here. Please clarify.

Reply:
@mikeloh77 you dont want to know 😛 but if you are curious, mainly because of [SelectedModel class in ModelSelector ](https://github.com/salesforce/TransmogrifAI/blob/c0d6ecb26ac8639a0934f5c98f84494d91acec23/core/src/main/scala/com/salesforce/op/stages/impl/selector/ModelSelector.scala#L235)